### PR TITLE
CORTX-29213 : Update k8s-cortx jenkins job for rgw based deployment

### PR DESCRIPTION
**Pull request by gauravchaudhari02**
Signed-off-by: Gaurav Chaudhari <gaurav.chaudhari@seagate.com>

# Problem Statement
- cortx-k8s Jenkins job is running on S3 way of deployment 

# Design
-  Shift deployment method to RGW based deployment

# Coding
   Checklist for Author
-  [x] Coding conventions are followed and code is consistent

# Testing 
  Checklist for Author
- [ ] Unit and System Tests are added
- [x] Test Cases cover Happy Path, Non-Happy Path and Scalability
   https://eos-jenkins.colo.seagate.com/job/Release_Engineering/job/re-workspace/job/gc-space/job/setup-k8s-cortx-cluster/6/
- [ ] Testing was performed with RPM

# Impact Analysis
  Checklist for Author/Reviewer/GateKeeper
- [ ] Interface change (if any) are documented
- [ ] Side effects on other features (deployment/upgrade)
- [ ] Dependencies on other component(s)

# Review Checklist 
  Checklist for Author
- [x] JIRA number/GitHub Issue added to PR
- [x] PR is self reviewed
- [x] Jira and state/status is updated and JIRA is updated with PR link
- [x] Check if the description is clear and explained

# Documentation
  Checklist for Author
- [ ] Changes done to WIKI / Confluence page / Quick Start Guide

### CODE REVIEW

File: jenkins/automation/kubernetes/deploy-cortx-k8-cluster.groovy [hunk 1]
```diff
@@ -20,10 +20,15 @@ pipeline {
...
         string(name: 'SNS_CONFIG', defaultValue: '1+0+0', description: 'sns configuration for deployment. Please select value based on disks available on nodes.', trim: true)
         string(name: 'DIX_CONFIG', defaultValue: '1+0+0', description: 'dix configuration for deployment. Please select value based on disks available on nodes.', trim: true)
+        string(name: 'CONTROL_EXTERNAL_NODEPORT', defaultValue: '31169', description: 'Port to be used for control service.', trim: true)
+        string(name: 'S3_EXTERNAL_HTTP_NODEPORT', defaultValue: '30080', description: 'HTTP Port to be used for IO service.', trim: true)
+        string(name: 'S3_EXTERNAL_HTTPS_NODEPORT', defaultValue: '30443', description: 'HTTPS to be used for IO service.', trim: true)
         text(defaultValue: '''hostname=<hostname>,user=<user>,pass=<password>''', description: 'VM details to be used. First node will be used as Primary node', name: 'hosts')
+        // Please configure CORTX_SCRIPTS_BRANCH and CORTX_SCRIPTS_REPO parameter in Jenkins job configuration.
```
Comment (shailesh-vaidya) [inline jenkins/automation/kubernetes/deploy-cortx-k8-cluster.groovy:31]: Need to add `EXTERNAL_EXPOSURE_SERVICE` choice parameter also.  Refer - https://github.com/Seagate/cortx-re/blob/main/jenkins/automation/kubernetes/setup-cortx-cluster.groovy#L34

**Review by shailesh-vaidya — CHANGES_REQUESTED** [review]
Please update - https://seagate-systems.atlassian.net/wiki/spaces/PRIVATECOR/pages/733479311/CORTX+Deployment+in+Kubernetes+Cluster accordingly.

File: jenkins/automation/kubernetes/deploy-cortx-k8-cluster.groovy (hunk 1 above)
Comment (gauravchaudhari02) [reply jenkins/automation/kubernetes/deploy-cortx-k8-cluster.groovy:31]: Done and tested

Comment (shailesh-vaidya) [reply jenkins/automation/kubernetes/deploy-cortx-k8-cluster.groovy:31]: I still do not see choice parameter at - https://github.com/gauravchaudhari02/cortx-re/blob/fix-k8s-cortx-job/jenkins/automation/kubernetes/deploy-cortx-k8-cluster.groovy#L30

**Review by shailesh-vaidya — CHANGES_REQUESTED** [review]

File: jenkins/automation/kubernetes/deploy-cortx-k8-cluster.groovy (hunk 1 above)
Comment (gauravchaudhari02) [reply jenkins/automation/kubernetes/deploy-cortx-k8-cluster.groovy:36]: Done. Added parameter in code.

**Review by shailesh-vaidya — APPROVED** [review]

**Review by shailesh-vaidya — APPROVED** [review]

**Pull request merged**